x-pack/plugins/ml/public/explorer/explorer.js

```diff
@@ -226,6 +237,11 @@ export const Explorer = injectI18n(injectObservablesAsProps(
             swimlaneViewByFieldName: currentSwimlaneViewByFieldName
           };
 
+          if (filterData.influencersFilterQuery !== undefined) {
+            Object.assign(stateUpdate, { ...filterData });
+          }
+
+          this.getIndexPattern(selectedJobs, stateUpdate.noInfluencersConfigured);
```

Comment:
I agree on moving indexPattern to the state since it is changing any time the job selection changes so it makes sense to keep it there and keep it in the React lifecycle. 
Updated in: 7032a0616ed061e43586af7f446ca00bbf237d5c